pandas/tests/io/test_parquet.py

```diff
@@ -712,6 +727,13 @@ def test_timestamp_nanoseconds(self, pa):
         df = pd.DataFrame({"a": pd.date_range("2017-01-01", freq="1n", periods=10)})
         check_round_trip(df, pa, write_kwargs={"version": "2.0"})
 
+    @td.skip_if_no("pyarrow", min_version="0.14")
+    def test_timezone_aware_index(self, pa, timezone_aware_date_list):
+        idx = 5 * [timezone_aware_date_list]
+
+        df = pd.DataFrame(index=idx)
```

Comment:
Can you also add it as a column (can use the same values) ?

Reply:
Nice catch, actually this assertion fails:
```
AssertionError: Attributes of DataFrame.iloc[:, 0] (column name="index_as_col") are different

Attribute "dtype" are different
[left]:  datetime64[ns, UTC-02:15]
[right]: datetime64[ns, pytz.FixedOffset(-135)]
```

Reply:
@jorisvandenbossche now that pyarrow does something unexpected I'm not sure how to proceed with this PR. Can you advice, please?

Reply:
It's not necessarily unexpected from pyarrow (it just always uses pytz (for better or worse) for fixed offsets, it only stores the actual fixed offset, not what class was used on the python side, but will answer about that on the Arrow PR). So I would just test the current behaviour: you can create the expected dataframe (with the different timezone) and pass this to `check_round_trip`. 

Separately from this, I am wondering if we should treat fixed offset `datetime.timezone(..)` and `pytz.FixedOffset(..)` as equal when comparing dtypes (@jbrockmendel ?). Or at least have an utility or keyword in the `assert_` to indicate they can be considered equal.

Reply:
@jorisvandenbossche I exposed the `check_dtype` parameter for now, this way it passes. What do you think?